src/eduid_common/api/helpers.py

```diff
@@ -58,6 +58,36 @@ def add_nin_to_user(user, proofing_state):
         current_app.logger.info('Sync result for user {!s}: {!s}'.format(proofing_user, result))
 
 
+def rm_nin_from_user(user, nin):
+    """
+    :param user: Central userdb user
+    :param nin: NIN to remove
+
+    :type user: eduid_userdb.user.User
+    :type nin: str
+
+    :return: None
+    """
+    proofing_user = ProofingUser(data=user.to_dict())
+    if proofing_user.nins.find(nin):
+        current_app.logger.info('Removing NIN {} for user {}'.format(nin, user))
+        proofing_user.nins.remove(nin)
+        proofing_user.modified_ts = True
+        # Save user to private db
+        current_app.proofing_userdb.save(proofing_user, check_sync=False)
+
+        proofing_state = current_app.proofing_statedb.get_state_by_eppn(proofing_user.eppn, raise_on_missing=False)
```

Comment:
I think we should check if the user has a state with another NIN in the proofing apps and replace that state with a new one instead of coupling removal of state and NIN.